portsyncd/linksync.cpp

```diff
@@ -90,15 +90,43 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
                        nlmsg_type, key.c_str(), admin, oper, addrStr, ifindex, master);
     }
 
-    /* Insert or update the ifindex to key map */
-    m_ifindexNameMap[ifindex] = key;
-
     /* teamd instances are dealt in teamsyncd */
     if (type && !strcmp(type, TEAM_DRV_NAME))
     {
         return;
     }
 
+    /* In the event of swss restart, it is possible to get netlink messages during bridge
+     * delete, interface delete etc which are part of cleanup. These netlink messages for
+     * the front-panel interface must not be published or it will update the statedb with
+     * old interface info and result in subsequent failures. A new interface creation shall
+     * not have master or admin status iff_up. So if the first netlink message comes with these
+     * values set, it is considered to be happening during a cleanup process.
+     * Fix to ignore this and any further messages for this ifindex
+     */
+
+    static std::map<unsigned int, std::string> m_ifindexOldNameMap;
```

Comment:
no need for `std::`

Reply:
Why use static variable, can you add it as a member attribute of LinkSync?

When would m_ifindexOldNameMap be cleared? If it never gets cleared, will it stop working for the second restart?

I suggest you use unordered_map to get O(1) look up time.